rules-reviewed/jboss-eap/websphere/WebSphereMQ.windup.xml

```diff
@@ -0,0 +1,1363 @@
+                    <message>Replace with Java EE 6 JMS: `javax.jms.TopicSession`.</message>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="websphere-mq-74ugps">
+            <when>
+                <javaclass references="com.ibm.mq.jms.MQXASession"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ API." description="A more general, and recommended, approach is to use the XASession class.">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/package-summary.html"></link>
+                </classification>
+                <hint effort="1">
+                    <message>Replace with Java EE 6 JMS: `javax.jms.XASession`.</message>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="websphere-mq-76hgq">
+            <when>
+                <javaclass references="com.ibm.mq.jms.MQXAQueueSession"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ API." description="A more general, and recommended, approach is to use the XAQueueSession class.">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/package-summary.html"></link>
+                </classification>
+                <hint effort="1">
+                    <message>Replace with Java EE 6 JMS: `javax.jms.XAQueueSession`.</message>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="websphere-mq-78hhew">
+            <when>
+                <javaclass references="com.ibm.mq.jms.MQXATopicSession"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ API." description="A more general, and recommended, approach is to use the XATopicSession class.">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/package-summary.html"></link>
+                </classification>
+                <hint effort="1">
+                    <message>Replace with Java EE 6 JMS: `javax.jms.XATopicSession`.</message>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+
+
+        <!-- SIB JMS -->
+        <rule id="websphere-mq-7Apvvd">
```

Comment:
Something like this:

         <rule id="websphere-mq-50000">
            <when>
                <javaclass references="{prefix}{type}"/>
            </when>
            <perform>
                <hint effort="1">
                    <message>Migrate to: javax.jms.{type}</message>
                    <link ... />
                </hint>
            </perform>
            <where param="prefix"><matches pattern="(com.ibm.websphere.sib.api.jms|com.ibm.msg.client.jms|com.ibm.jms)(Jms|JMS)?+"/></where>
        </rule>

This should also have doc links.